test/Integration/ApplicationTrait.php

```diff
@@ -0,0 +1,72 @@
+<?php
+/**
+ * @see       https://github.com/zendframework/zend-mvc-middleware for the canonical source repository
+ * @copyright Copyright (c) 2019 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license   https://github.com/zendframework/zend-mvc-middleware/blob/master/LICENSE.md New BSD License
+ */
+
+namespace ZendTest\Mvc\Middleware\Integration;
+
+use Zend\Mvc\Application;
+use Zend\Mvc\MvcEvent;
+use Zend\Mvc\SendResponseListener;
+use ZendTest\Mvc\Middleware\Integration\TestAsset\NoopSendResponseListener;
+
+trait ApplicationTrait
+{
+    /**
+     * @var Application
+     */
+    protected $application;
+
+    /**
+     * Extra config to use during application set up
+     */
+    protected $extraConfig = [];
+    /**
+     * Fail test with exception message if mvc error event is triggered.
+     */
+    protected $failOnErrorEvents = true;
```

Comment:
I can't see anywhere that we set it to `false`. Maybe it can be removed then?

Reply:
It is a crutch to make tests fail with verbose information containing actual error. It could get in a way if error flow will be tested in integration tests so there is this flag to disable them selectively.

I expect it to get useful later. I would like to keep it until used or some general cleanup work later.